qiskit/visualization/latex.py

```diff
@@ -235,44 +220,35 @@ def _initialize_latex_array(self):
                 self._latex[i][0] = label
 
     def _get_image_depth(self):
-        """Get depth information for the circuit.
-
-        Returns:
-            int: number of columns in the circuit
-            int: total size of columns in the circuit
-        """
-
+        """Get depth information for the circuit."""
         max_column_widths = []
-        # Determine row spacing before image depth
+        # Determine wire spacing before image depth
         for layer in self.ops:
             for op in layer:
-                # useful information for determining row spacing
-                boxed_gates = ['u0', 'u1', 'u2', 'u3', 'x', 'y', 'z', 'h', 's',
-                               'sdg', 't', 'tdg', 'rx', 'ry', 'rz', 'ch', 'cy',
-                               'crz', 'cu3', 'id']
+                # useful information for determining wire spacing
+                boxed_gates = ['u1', 'u2', 'u3', 'u', 'p', 'x', 'y', 'z', 'h',
+                               's', 'sdg', 't', 'tdg', 'sx', 'sxdg', 'rx', 'ry', 'rz',
+                               'ch', 'cy', 'crz', 'cu2', 'cu3', 'cu', 'id']
                 target_gates = ['cx', 'ccx', 'cu1', 'cp', 'rzz']
                 if op.name in boxed_gates:
                     self.has_box = True
-                if op.name in target_gates:
-                    self.has_target = True
-                if isinstance(op.op, ControlledGate):
+                elif op.name in target_gates:
                     self.has_target = True
+                elif isinstance(op.op, ControlledGate):
+                    self.has_box = True
 
         for layer in self.ops:
 
-            # store the max width for the layer
             current_max = 0
-
             for op in layer:
-
-                # update current op width
                 arg_str_len = 0
 
                 # the wide gates
                 for arg in op.op.params:
-                    arg_str = re.sub(r'[-+]?\d*\.\d{2,}|\d{2,}',
-                                     _truncate_float, str(arg))
-                    arg_str_len += len(arg_str)
+                    if not any(isinstance(param, np.ndarray) for param in op.op.params):
+                        arg_str = re.sub(r'[-+]?\d*\.\d{2,}|\d{2,}',
+                                         self._truncate_float, str(arg))
```

Comment:
They way to formatting the complex number here is hard to read (you know how much I would like to avoid regexs. I think a combination of `round` (for `_truncate_float`) and `arg.real, arg.imag` can do the trick here.

Reply:
The _get_image_depth method is computing in a very large ballpark. Before I added the check to not include arrays, the old version included them, which meant some layers were computed to be 100+ chars wide, even though they only displayed a few chars wide. 

Also these lines are assuming that params display as floats, which ignores pi_check. If a param is pi/2, this will compute len(arg_str) as 6, even though the Latex fraction is only 1 char wide.

So I believe we should leave this alone and someone should do a deep dive into this method to see how image depth is being computed and how it can be simplified and made more accurate.

Even with the above issues, things seem to display correctly, so not sure what's going on. It may be that the qcircuit and xy-pic commands are overriding the beamer settings. For a future issue.